### ISSUE TITLE
Container has type of first generic type parameter only

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#483]
**Describe the bug**
When extracting the type of parametrised generics inside a comprehension, the result is assigned to an iterable with the parameter of the first item in the sequence only.

**To Reproduce**
Type check the snippet below and observe that the type is `Generator[int]`.

**Expected behavior**
That the revealed type would be `Generator[Union[int, str]]`.

**Screenshots or Code**
```py
from typing import Generic, TypeVar

T = TypeVar('T')

class Bar(Generic[T]):
    def baz(self) -> T: ...


bar = [Bar[int](), Bar[str]()]
reveal_type((b.baz() for b in bar))
```

**VS Code extension or command-line**
extension, 1.1.19

**Comment by erictraut** [comment]
Thanks for the bug report. This was a bug in the logic that deduplicates union subtypes. It wasn't taking into account the specialized return type for methods, so it incorrectly eliminated one of the two union types for the expression `b.baz`. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now fixed in 1.1.20, which I just published.
